app/api/groups/update_group.go

```diff
@@ -0,0 +1,105 @@
+package groups
+
+import (
+	"errors"
+	"net/http"
+	"time"
+
+	"github.com/go-chi/render"
+
+	"github.com/France-ioi/AlgoreaBackend/app/database"
+	"github.com/France-ioi/AlgoreaBackend/app/service"
+)
+
+type groupUpdateInput struct {
+	// Nullable fields are of pointer types
+	Type          string     `json:"type" sql:"column:sType" valid:"in(Class|Team|Club|Friends|Other)"`
+	Name          string     `json:"name" sql:"column:sName"`
+	Grade         int64      `json:"grade" sql:"column:iGrade"`
+	Description   *string    `json:"description" sql:"column:sDescription"`
+	Opened        bool       `json:"opened" sql:"column:bOpened"`
+	FreeAccess    bool       `json:"free_access" sql:"column:bFreeAccess"`
+	PasswordTimer *string    `json:"password_timer" sql:"column:sPasswordTimer" valid:"matches(^\\d{2}:[0-5]\\d:[0-5]\\d$)"`
```

Comment:
I'm not sure. The time.Duration type stores nanoseconds and cannot parse this kind of data. We could implement custom marshalling here, but we do not use this value, we just transfer it into the DB.

Reply:
It would be the opportunity to make the UI use a more appropriate format as well. Ok we should keep that for later (in the DB backlog, using a number directly).